plutus-ledger/src/Ledger/Index.hs

```diff
@@ -333,7 +333,7 @@ matchInputOutput txid mp txin txo = case (txInType txin, txOutDatumHash txo, txO
         unless (validatorHash v == vh) $ throwError $ InvalidScriptHash v vh
 
         pure $ ScriptMatch (txInRef txin) v r d
-    (Just ConsumePublicKeyAddress, Nothing, Address{addressCredential=PubKeyCredential pkh}) ->
+    (Just ConsumePublicKeyAddress, _, Address{addressCredential=PubKeyCredential pkh}) ->
```

Comment:
Note that we have already patch the simulator so as to discard datum hash present in pubkey utxo (see function `lkpTxOut` )